src/collector.ts

```diff
@@ -132,10 +130,18 @@ export async function runCollector(
         let blendPoolEvent = poolEventFromEventResponse(raw_event);
         if (blendPoolEvent) {
           // handle pool events immediately
+          let poolConfig = poolConfigs.find(
+            (config) => config.poolAddress === blendPoolEvent.contractId
+          );
+          if (!poolConfig) {
+            logger.error(`Pool config not found for event: ${stringify(blendPoolEvent)}`);
+            continue;
+          }
```

Comment:
IMO the responsibility for if an event should be dropped or not is not really part of this codes responsibility.

I'd send all events we get from the query if we are able to parse an event from it, and just supply the `blendPoolEvent.contractid`, and allow the `poolEventHandler` to deal with the case where (by some magical reason?) we get an event from a contract we did not filter for